pin-project-internal/src/pin_project/enums.rs

```diff
@@ -34,17 +34,23 @@ pub(super) fn parse(cx: &mut Context, mut item: ItemEnum) -> Result<TokenStream>
     let Context { proj_ident, proj_trait, orig_ident, lifetime, .. } = &cx;
     let proj_generics = cx.proj_generics();
     let proj_ty_generics = proj_generics.split_for_impl().1;
-    let (impl_generics, ty_generics, where_clause) = item.generics.split_for_impl();
+
+    let ProjTraitGenerics { impl_generics, ty_generics, where_clause, orig_ty_generics } =
+        cx.proj_trait_generics();
 
     let mut proj_items = quote! {
         #[allow(dead_code)]
         enum #proj_ident #proj_generics #where_clause { #(#proj_variants,)* }
     };
+
+    let crate_path = crate::utils::crate_path();
+
     proj_items.extend(quote! {
-        impl #impl_generics #proj_trait #ty_generics for ::core::pin::Pin<&mut #orig_ident #ty_generics> #where_clause {
-            fn project<#lifetime>(&#lifetime mut self) -> #proj_ident #proj_ty_generics #where_clause {
+        impl #impl_generics #proj_trait #ty_generics for ::core::pin::Pin<&#lifetime mut #orig_ident #orig_ty_generics> #where_clause {
+            fn project(&mut self) -> #proj_ident #proj_ty_generics #where_clause {
```

Comment:
```suggestion
            fn project(&mut self) -> #proj_ident #proj_ty_generics {
```
Preexisting, but as it is already declared in the impl block, it looks unnecessary.